### PR TITLE
feat(drawer): L3-3071 added drawer component

### PR DESCRIPTION
**Pull request by constantinehuzenko**
**Jira ticket**

[Jira Ticket ID](https://phillipsauctions.atlassian.net/browse/L3-3071)

**Screenshots**
| Before | After |
| ------ | ----- |
| image | image |

**Summary**

Added drawer component using Radix Primitivis  

**Change List (describe the changes made to the files)**

- Additions, Changes, and Radix primitives

**Acceptance Test (how to verify the PR)**

- verify that drawer opens on button click
- verify that content of drawer visible
- verify that drawer full size on phone
- verify that drawer closes hitting both: overlay and closing icon 

**Regression Test**

- (Optional) Add verification steps to make sure this PR doesn't break old functionality

**Evidence of testing**

- image

**Things to look for during review**

- [ ] PR title should correctly describe the most significant type of commit. I.e. `feat(scope): ...` if a `minor` release should be triggered.
- [ ] All commit messages follow convention and are appropriate for the changes
- [ ] All references to `phillips` class prefix are using the prefix variable
- [ ] All major areas have a `data-testid` attribute.
- [ ] Document all props with jsdoc comments
- [ ] All strings should be translatable.
- [ ] Unit tests should be written and should have a coverage of 90% or higher in all areas.

New Components

- [ ] Are there any [accessibility considerations](https://www.w3.org/WAI/ARIA/apg/patterns/) that need to be taken into account and tested?
- [ ] Default story called "Playground" should be created for all new components
- [ ] Create a jsdoc comment that has an Overview section and a link to the Figma design for the component
- [ ] Export the component and its typescript type from the `index.ts` file
- [ ] Import the component scss file into the `styles.scss` file.

### CODE REVIEW

File: src/components/Drawer/Drawer.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { getCommonProps, noOp } from '../../utils';
+import classnames from 'classnames';
+import * as Dialog from '@radix-ui/react-dialog';
+import CloseIcon from '../../assets/close.svg?react';
+import IconButton from '../IconButton/IconButton';
+
+// You'll need to change the HTMLDivElement to match the top-level element of your component
+export interface DrawerProps extends React.HTMLAttributes<HTMLDivElement> {
+  isOpen?: boolean;
```
Comment (davidicus) [inline src/components/Drawer/Drawer.tsx:10]: interface needs some comments for docs

Comment (constantinehuzenko) [reply src/components/Drawer/Drawer.tsx:10]: @davidicus can I add it in new Footer's PR? so we can merge this one and I can finish Footer today

**Review by davidicus — CHANGES_REQUESTED** [review]
Looks great. Just a few small comments.

File: src/components/Drawer/_drawer.scss [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    border-radius: 100%;
+    color: $pure-black;
+    display: inline-flex;
+    justify-content: center;
+    position: absolute;
+    right: 10px;
+    top: 10px;
+  }
+
+  @keyframes overlay-show {
```
[the remark this answers is not in the document]
Comment (davidicus) [reply src/components/Drawer/_drawer.scss:45]: This is my bad. I know why it is done with animation now. It looks like it is only added to the DOM when it opens, which is good. But that is preventing the transitions from being applied. I will resolve this comment.

**Review by davidicus — APPROVED** [review]

**Pull request merged**